adal/src/telemetry/java/com/microsoft/aad/adal/APIEvent.java

```diff
@@ -0,0 +1,172 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+import android.content.Context;
+import android.util.Pair;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class tracks flows for certain API calls. Most notably in those all acquireToken* calls
+ * All other Events will be called within the timeline of APIEvent
+ */
+class APIEvent extends DefaultEvent {
+
+    private static final String TAG = DefaultEvent.class.getSimpleName();
+    private final String mEventName;
+
+    APIEvent(final String eventName) {
+        setEvent(EventStrings.EVENT_NAME, eventName);
+        mEventName = eventName;
+    }
+
+    APIEvent(final String eventName, final Context context, final String clientId) {
+        this(eventName);
+        setDefaults(context, clientId);
+    }
+
+    void setAuthority(final String authority) {
+        if (StringExtensions.isNullOrBlank(authority)) {
+            return;
+        }
+
+        setEvent(EventStrings.AUTHORITY_NAME, authority);
+        final URL authorityUrl = StringExtensions.getUrl(authority);
+        if (authorityUrl == null) {
+            return;
+        }
+
+        if (UrlExtensions.isADFSAuthority(authorityUrl)) {
+            setAuthorityType(EventStrings.AUTHORITY_TYPE_ADFS);
+        } else {
+            setAuthorityType(EventStrings.AUTHORITY_TYPE_AAD);
+        }
+    }
+
+    void setAuthorityType(final String authorityType) {
+        setEvent(EventStrings.AUTHORITY_TYPE, authorityType);
+    }
+
+    void setIsDeprecated(final Boolean isDeprecated) {
+        setEvent(EventStrings.API_DEPRECATED, isDeprecated.toString());
+    }
+
+    void setValidationStatus(final String validationStatus) {
+        setEvent(EventStrings.AUTHORITY_VALIDATION, validationStatus);
+    }
+
+    void setExtendedExpiresOnSetting(final Boolean extendedExpiresOnSetting) {
+        setEvent(EventStrings.EXTENDED_EXPIRES_ON_SETTING, extendedExpiresOnSetting.toString());
+    }
+
+    void setSilentRequestPromptBehavior(final String promptBehavior) {
+        setEvent(EventStrings.PROMPT_BEHAVIOR, promptBehavior);
+    }
+
+    void setAPIId(final String id) {
+        setEvent(EventStrings.API_ID, id);
+    }
+
+    String getEventName() {
+        return mEventName;
+    }
+
+    void setWasApiCallSuccessful(final Boolean isSuccess, final Exception exception) {
```

Comment:
ah? We're saving the string value, and you can always do String.valueOf(). Generally we shoud avoid using the Boolean and I also don't see any reason why we are using the Boolean.